src/Workspaces/Core/Portable/Utilities/AbstractSpeculationAnalyzer.cs

```diff
@@ -521,6 +526,40 @@ private bool ReplacementChangesSemanticsForNode(SyntaxNode currentOriginalNode,
             return false;
         }
 
+        /// <summary>
+        /// Determine if removing the cast could cause the semantics of System.Object method call.
```

Comment:
You may want to put a comment inside the method somewhere just to motivate this example a bit.